src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java

```diff
@@ -0,0 +1,160 @@
+package com.microsoft.azure.cosmosdb.kafka.connect.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.kafka.connect.data.Values.convertToByte;
+import static org.apache.kafka.connect.data.Values.convertToDouble;
+import static org.apache.kafka.connect.data.Values.convertToFloat;
+import static org.apache.kafka.connect.data.Values.convertToInteger;
+import static org.apache.kafka.connect.data.Values.convertToLong;
+import static org.apache.kafka.connect.data.Values.convertToShort;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class JsonToStruct {
```

Comment:
Sonarlint is reporting linter issues on line 30 and the lines with `logger.error`.

Reply:
Didn't check earlier but sonarlint is finding issues with the fillrecords method in the sourcetask class